retriever/lib/scripts.py

```diff
@@ -1,36 +1,86 @@
-import csv
+from __future__ import print_function
+
+from future import standard_library
+
+from retriever.lib.tools import open_fr
+
+standard_library.install_aliases()
 import imp
-import io
 import os
 import sys
-from os.path import join, isfile, getmtime, exists
+import urllib.request
+import urllib.parse
+import urllib.error
+from os.path import join, isfile, getmtime, exists, abspath
 
 from pkg_resources import parse_version
 
+from retriever.lib.defaults import SCRIPT_SEARCH_PATHS, VERSION, SCRIPT_WRITE_PATH
 from retriever.lib.compile import compile_json
-from retriever.lib.defaults import SCRIPT_SEARCH_PATHS, VERSION, ENCODING
+
+global_temp_scripts = {}
+
+def check_retriever_minimum_version(module):
+    mod_ver = module.retriever_minimum_version
+    m = module.name
+    if not parse_version(VERSION) >= parse_version("{}".format(mod_ver)):
+        print("{} is supported by Retriever version ""{}".format(m, mod_ver))
+        print("Current version is {}".format(VERSION))
+        return False
+    return True
 
 
 def MODULE_LIST(force_compile=False):
     """Load scripts from scripts directory and return list of modules."""
+    if not force_compile and global_temp_scripts:
+        return global_temp_scripts._getScripts()
```

Comment:
I think you've got force_compile right. Renaming it to reload_json_scripts would be clearer now that we are no longer compiling scripts in the same sense.  

As for reloading the json scripts in a single session, I didn't have a specific case in mind. I left it there just in case we would ever want to force compile the scripts in the same session, since I wasn't 100% sure that would never happen.

Reply:
Ok that makes sense! I think Henry's doing a full review on this PR again, so I'll be sure to take it out in the next update.